src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/PlayerDeathSystem.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.ligthandshadow.componentsystem.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+import org.terasology.ligthandshadow.componentsystem.components.LASTeamComponent;
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.characters.CharacterTeleportEvent;
+import org.terasology.logic.health.BeforeDestroyEvent;
+import org.terasology.logic.health.DoHealEvent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+
+
+@RegisterSystem
+public class PlayerDeathSystem extends BaseComponentSystem {
+    Logger logger = LoggerFactory.getLogger(PlayerDeathSystem.class);
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_HIGH)
+    public void beforeDestroy(BeforeDestroyEvent event, EntityRef player, CharacterComponent characterComponent, AliveCharacterComponent aliveCharacterComponent) {
+        if (player.hasComponent(PlayerCharacterComponent.class)) {
+            event.consume();
+            logger.info(player.toFullDescription());
+            String team = player.getComponent(LASTeamComponent.class).team;
+            player.send(new DoHealEvent(100000, player));
+            player.send(new CharacterTeleportEvent(LASUtils.getTeleportDestination(team)));
```

Comment:
Ideally, we should not be using the exact location of the base. That would mean when multiple players die, they always land on top of each other. Since the base is big enough, we could teleport to a random position on the base as in `LASUtils.getTeleportDestination(team)) + Vec3(random(-1, 1), random(-1, 1), 0)`.

Again, here make sure that the players always face the opponent team's base. Maybe extract the above completely into a method and use it in the team selection logic as well.